docs/snippets/proc-building-function.md

```diff
@@ -0,0 +1,43 @@
+<!-- Snippet used in the following topics:
+- /docs/getting-started/build-run-deploy-func.md
+- /docs/functions/build-run-deploy-func.md
+-->
+The `build` command uses the image name and the image registry to build a container image for your function.
+
+The `func.yaml` file is read to determine the image name and registry. If the function project has not previously been built, either the registry or the image name must be provided, and the image name is stored in the configuration `func.yaml` file.
```

Comment:
```suggestion
If the function project has not previously been built, you will be prompted to provide an image registry.
```
The user __may__ specify the registry (or image) on the command line. However, we have improved the UX so that if it is not known (either because it was not specified via `--registry` or the `FUNC_REGISTRY` environment variable is unset), the user will be prompted. E.g.

```
❯ ./func build -p foobar
A registry for function images is required. For example, 'docker.io/tigerteam'.
? Registry for function images:
```

Also, I'm not sure we really need to surface `func.yaml` yet. 🤷